### ISSUE TITLE
Do local variables for scopes post lifting 

### ISSUE THREAD

**Issue by laurenarnett** [laurenarnett/TensorFlock#138]
```
transpose : T<n,m> -> T<m,n>;
transpose mat = mat';
{
mat':T<m,n>;
mat'[j,i] = mat[i,j];
}
```
* allocate args
* allocate stuff in scope

**Comment by laurenarnett** [comment]
* topsort variables
* build alloca's for them inside of codegen function body

**Comment by laurenarnett** [comment]
Proving to be less trivial than expected ☹️
Found bug when trying to find scope of a variable. Somehow semant is not finding that there is a scope for a variable in a case such as 
```
main = f;
f : Nat;
f = a + b; {
a : Nat;
a = 1;
b : Nat;
b = 1
}
```

**Comment by laurenarnett** [comment]
This is the fault of topsort

**Comment by laurenarnett** [comment]
solved with #141
